packages/cardhost/app/templates/components/cards/cardstack/base-card/embedded.hbs

```diff
@@ -3,7 +3,7 @@
 {{!-- TODO use a sanitized version of the @card.id in the class name after we have css coming from card documents --}}
 <div
   data-test-embedded-card={{@card.name}}
-  class="embedded-card cardstack_base-card"
+  class="embedded-card cardstack_base-card {{@cardId}} {{@mode}}"
```

Comment:
this isn't going to do the thing that you want. what you _actually_ want is to add the adopted card's ID into the template--otherwise this is going to be a totally different class for every card that exists and there will be no commonality.

Instead you should do this:
```hbs
class="embedded-card {{if @card.adoptedFrom @card.adoptedFrom.name "cardstack_case-card"}} {{@mode}}"
```

Reply:
I added `@card.name` in there too because you'll also need to style the template to look like the cards that adopt from the template. so the template itself will need to use the `@card.name` css selector (use the card name, not the id--that way you can have consistent styling across realms when we start supporting that). and then all the cards that adopt from the card name need to be styled exactly the same, so you should also add a `adopts-@card.name` css selector to the same styles that you create for the template.

you should use the card adoption mechanism to create some event cards and you'll see what I mean.

Reply:
Also, let's hoist this class to the card renderer so that the user cant break this--as the card template will eventually be edited by the user.

Reply:
let's also make a `(safe-string)` helper too so that you can make safe CSS strings from the card names. this way you don't need to pass the template name around everywhere (and also template is probably the wrong name for this).

Reply:
so maybe that all looks like:
```hbs
class="embedded-card {{if @card.adoptedFrom (safe-string (concat "adopts-" @card.adoptedFrom.name)) "cardstack_base-card"}} {{@card.name}} {{@mode}}"

```

Reply:
If you're not familiar with creating a helper, they are really quite simple. Here is an example of a helper in the core package that adds `1` to the value: https://github.com/cardstack/cardstack/blob/master/packages/core/addon/helpers/inc.js

so you'd use it like:
```hbs
<div>(inc @field.position)</div> 
```
and it would spit out what ever the field's position is plus 1.